### ISSUE TITLE
Retroball doesn't work.

### ISSUE THREAD

**Issue by Keatah** [AppleWin/AppleWin#298]
Retroball freezes up. I bet it has something to do with hardware vertical sync mod that shipped with the game. You had a choice to use the wire, or not. You just gave up some smoothness if you didn't.

ftp://ftp.apple.asimov.com/pub/apple_II/images/games/file_based/juggler_retro-ball_skeetshoot.dsk

**Comment by sicklittlemonkey** [comment]
Yeah, it's stuck reading $C060.

This code in Tape.cpp:
return (1<<7) | (MemReadFloatingBus(nCyclesLeft) & 0x7F); // Keep high-bit
fixed (since TAPEIN isn't supported)

... (while also showing a lack of knowledge about bit manipulation in C)
should probably be something like:
return MemReadFloatingBus(0, nCyclesLeft); // high-bit clear when TAPEIN
not connected

I'm assuming that's what real hardware does.

Cheers,
Nick.

**Comment by sicklittlemonkey** [comment]
> I'm assuming that's what real hardware does.

Ah, assumptions ... just tested on a PAL //e.

$C060 returns high bit set with nothing connected to the cassette input.

My testing is limited to that for now. Can anyone else test?

BTW, in Memory.cpp in:
static BYTE __stdcall IORead_C06x(WORD pc, WORD addr, BYTE bWrite, BYTE d,
ULONG nCyclesLeft)

switch (addr & 0xf)
should be:
switch (addr & 0x7) // address bit 4 is ignored (UTA2E page 7-5)
and cases 0x8-0xF can be deleted.

It would be good to standardize references like the above. Perhaps add a
comment at the top after the Author section:
 *
 * Author: Various
 *
 * In comments, UTA2E is an abbreviation for a reference to "Understanding
the Apple //e" by James Sather
 */

Cheers,
Nick.

**Comment by tomcw** [comment]
Hi Nick,

I forced $C060's bit7 high because of #115.

>$C060 returns high bit set with nothing connected to the cassette input.

So I guess this is because of pull-up resistor R9 (again see #115)?

**Comment by sicklittlemonkey** [comment]
No, I think that resistor only affects the TAPEOUT voltage level.

If you look at Sather's description of the TAPEIN circuit amplifier (UTA2E page 7-5 items 1, 2, 3) you'll see that it basically inverts the signal. He doesn't state that explicitly, but see the diagram on page 7-6.

**Comment by tomcw** [comment]
NB. Nick's comment in 37b3006f5c7dc5dcef76ab4e285b92a425038a34 is:

>TAPEIN has high bit 1 when input is low or not connected (UTA2E page 7-5, 7-6)

----

I haven't tried the image, but do we know definitely that the game works without the Video Sync connector?

btw. I found this picture on ebay:
![retroball](https://cloud.githubusercontent.com/assets/6696896/8577021/d6222c9a-259d-11e5-9d4a-1b6d5b66164a.JPG)

**Comment by sicklittlemonkey** [comment]
> I haven't tried the image, but do we know definitely that the game works
without the Video Sync connector?

I don't see how it can work without it. But we don't have the original
image - just the file linked to - maybe the original let you choose.

Today I tested an NTSC //e and a II europlus. Both have $C060 high. I can't
transfer an image right now - wish I had an SDFloppy or Floppy Emu.

My CFFA 3000 lives in my IIgs ... Hopefully someone else can test the image
on real hardware, but it will surely just hang.

Cheers,
Nick.

**Comment by Keatah** [comment]
I tested the game on a Platinum //e and standard II+. It requires the videosync mod to be used. You know it's working right away because the title on the playfield scrolls.

If you disconnect the mod mid-game it sometimes stops working. Sometimes not. And reconnecting it doesn't let it run again, if it stopped. You have to reboot and restart the game.

Note (on my II+) I needed to reverse the orange connector from left to right, as opposed to what was in the manual. I suppose the difference is just the mainboard revision or a typo in the manual.

On the Platinum //e I used the single stake connector near the 4-pin RF modulator connector. That's the video out pin.

I suppose you can do the same mod by tapping into the RCA-video connector with a pass-through-tap custom cable. A matter of aesthetics and cost, to be sure.

**Comment by Keatah** [comment]
As a kid in the 70's and 80's I could swear this gamed worked without that wire. The only thing I could think of is I had another card in my system at the time, or a different revision of the game. I swear it!

**Comment by Keatah** [comment]
It's also possible I had so much crap hooked up to my machine at the time that video noise somehow came in on the cassette connector and that allowed me to play without the wire mod.
